src/Lecture1.hs

```diff
@@ -104,7 +134,16 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Int
+strSum str = 
+    sum 0 (words str)
+    where
+        sum :: Int -> [String] -> Int
+        sum result numbers =
+            if null numbers
+                then result
+                else sum (result + (read (head numbers) :: Int)) (tail numbers)
```

Comment:
`sum` is already a standard function in Haskell, you can use it instead.

Also, as you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
                else sum (result + (read (head numbers))) (tail numbers)
```

I see that the `result` variable is redundant in your implementation of `sum`. You don't change it so it's always `0` and doesn't affect the final outcome.

So if you use standard functions `map` and `sum`, you can implement `strSum` as a nice one-liner 🙂

Reply:
Done!